core/src/main/javascript/crypto/KeyFormat.js

```diff
@@ -30,22 +33,27 @@
         /**
          * Normalize public key input into expected WebCrypto format.
          *
-         * @param {string|JSON|Uint8Array} input Base64-encoded or JSON or ByteArray of key.
-         * @param {string} format key format type ("spki" | "jwk")
-         * @return webcrypto format for the public key.
+         * @param {string|object|Uint8Array} input Base64-encoded or JSON object or ByteArray of key.
+         * @param {KeyFormat} format key format type ("spki" | "jwk")
+         * @return WebCrypto acceptable format of the public key.
          * @throws MslCryptoException if the key data is invalid.
          */
         normalizePubkeyInput: function normalizePubkeyInput(input, format) {
             if (format == KeyFormat.SPKI) {
                 try {
-                    input = (typeof input == "string") ? base64$decode(input) : input;
+                    input = (typeof input === "string") ? base64$decode(input) : input;
                 } catch (e) {
                     throw new MslCryptoException(MslError.INVALID_PUBLIC_KEY, format + " " + input, e);
                 }
             }
             else if (format == KeyFormat.JWK) {
                 try {
-                    input = (typeof input == "string") ? JSON.parse(input) : input;
+                    input = (typeof input === "string") ? JSON.parse(input) : input;
+                    /* input must be a JSON object */
+                    if (typeof input !== "object") {
```

Comment:
Actually you also need to do both checks: `typeof x === "object" && x.constructor === Object`. Just typeof isn't good enough because it will also catch other things that are not pure JavaScript objects.